pkg/madmin/obd.go

```diff
@@ -259,19 +260,24 @@ func (adm *AdminClient) ServerOBDInfo(ctx context.Context, obdDataTypes []OBDDat
 			respChan <- OBDInfo{
 				Error: err.Error(),
 			}
+			close(respChan)
```

Comment:
It looks like it is easier to run defer close(respChan) in the beginning of this goroutine

Reply:
@wlan0 ^^

Reply:
There is a functional difference between closing it here and using a defer. Defer statements get called sometime in the future after the return. It is not immediate. 

So, there is a really important corner case where this matters. When minio is under load and say we start OBD, it will overload the system and start erroring out. In this scenario, we must prevent a cascading failure from overloading the system further, and stop further OBD tests. We do this by closing the channel, which will end up releasing this goroutine, and killing the connection to the server. This will finally release the OBD lock. Otherwise, the load will kill the servers, and they will go down while holding the OBD lock. You'll be left in a state where you can't debug for another 10 mins because the lock will be held. 

Explicit closing is also important for crisp UX on the client side. `mc` will keep spinning even after the error message is displayed because the channel will not be closed.

Reply:
@vadmeste ^^